src/Lecture3.hs

```diff
@@ -125,9 +135,14 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
+  (<>) :: Reward -> Reward -> Reward
+  a <> b = Reward { rewardGold = foldMap rewardGold [a, b]
+                  , rewardSpecial = foldr ((||) . rewardSpecial) False [a, b]
```

Comment:
If you want to write in the list-style, you can use the standard function `all` here 🙂 
```suggestion
                  , rewardSpecial = all rewardSpecial [a, b]
```

Reply:
I think it should be `any`